lib/product.service.ts

```diff
@@ -33,14 +39,59 @@ export class ProductStack extends Stack {
       }
     });
 
+    // DynamoDB tables
+    const productsTable = new dynamo.Table(this, 'ProductsTable', {
+      partitionKey: { name: 'id', type: dynamo.AttributeType.STRING },
+     });
 
+    const stockTable = new dynamo.Table(this, 'StockTable', {
+      partitionKey: { name: 'product_id', type: dynamo.AttributeType.STRING },
+    });
+
+    // Table names created on asw console
+    const PRODUCTS_TABLE_NAME = 'Products';
+    const STOCK_TABLE_NAME = 'Stock';
+
+    // DynamoDB lambdas
+    const getProductsListLambda = new NodejsFunction(this, 'get-products-list-lambda', {
+      functionName: 'getProductsList',
+      entry: path.resolve(__dirname, '../src/products/products-dynamodb.handler.ts'),
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: 'getProductsList',
+      environment: {
+        PRODUCTS_TABLE_NAME: PRODUCTS_TABLE_NAME,
+        STOCK_TABLE_NAME: STOCK_TABLE_NAME
+      }
+    });
+    
+    const getProductsByIdLambda = new NodejsFunction(this, 'get-products-by-id-lambda', {
+      functionName: 'getProductsById',
+      entry: path.resolve(__dirname, '../src/products/products-dynamodb.handler.ts'),
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: 'getProductsById',
+      environment: {
+        PRODUCTS_TABLE_NAME: PRODUCTS_TABLE_NAME,
+      }
+    });
+
+    const createProductLambda = new NodejsFunction(this, 'create-product-lambda', {
+      functionName: 'createProduct',
+      entry: path.join(__dirname, '../src/products/products-dynamodb.handler.ts'),
+      handler: 'createProduct',
+      runtime: lambda.Runtime.NODEJS_18_X,
+      environment: {
+        PRODUCTS_TABLE_NAME: PRODUCTS_TABLE_NAME
+      }
+    });
+
+    // api
     const productsApi = new RestApi(this, "products-api", {
       restApiName: "Products API Gateway",
       description: "This Products-API serves the Lambda functions."
     });
 
-    const productsFromLambdaIntegration = new LambdaIntegration(lambdaProductsList, {
-
+    // serverless integration
+    const productsFromLambdaIntegration = new LambdaIntegration(getProductsListLambda, {
       integrationResponses: [
```

Comment:
1) this object property "integrationResponses" is duplicated in 3 places at least. Could be created as one common const "prop" and then reused.  

2) Why do we need 
```
responseTemplates: {
            'application/json': "$util.parseJson($input.json('$.body'))"
          }
```
for the GET lambdas?

3) also you don't need the whole list of Allow-Methods here, as it is only GET

4) For the Allow-Origin, it is good to use env variable.

5) And i was not successful in non-proxy config, so can you pls tell me, how does your 400, 404, 500, other statusCodes work for you? are these status codes are properly propagated to the front-end when some error happens?